src/main/java/org/testng/remote/strprotocol/BaseMessageSender.java

```diff
@@ -124,61 +127,68 @@ public void initReceiver() throws SocketTimeoutException {
       serverSocket = new ServerSocket(m_port);
       serverSocket.setSoTimeout(5000);
 
-	  Socket socket = serverSocket.accept();
-	  m_inStream = socket.getInputStream();
-	  m_inReader = new BufferedReader(new InputStreamReader(m_inStream));
-	  m_outStream = socket.getOutputStream();
-	  m_outWriter = new PrintWriter(new OutputStreamWriter(m_outStream));
+      Socket socket = null;
+      while (!m_requestStopReceiver) {
+        try {
+          if (m_debug) {
+            p("polling the client connection");
+          }
+          socket = serverSocket.accept();
+          // break the loop once the first client connected
+          break;
+        }
+        catch (IOException ioe) {
+          try {
+            Thread.sleep(100L);
+          }
+          catch (InterruptedException ie) {
+            // Do nothing.
+          }
+        }
+      }
+      if (socket != null) {
+        m_inStream = socket.getInputStream();
+        m_inReader = new BufferedReader(new InputStreamReader(m_inStream));
+        m_outStream = socket.getOutputStream();
+        m_outWriter = new PrintWriter(new OutputStreamWriter(m_outStream));
+      }
     }
     catch(SocketTimeoutException ste) {
-      try {
-		serverSocket.close();
-	  } catch (IOException e) {
-		  // ignore
-	  }
       throw ste;
     }
     catch (IOException ioe) {
-      // TODO Auto-generated catch block
-      ioe.printStackTrace();
+      closeQuietly(serverSocket);
     }
   }
 
+  public void stopReceiver() {
```

Comment:
this is invoked at eclipse plugin side: https://github.com/cbeust/testng-eclipse/pull/166/files#r37427417